classes/class-wc-connect-taxjar-integration.php

```diff
@@ -689,11 +689,11 @@ public function get_taxable_address() {
 	 * @return array
 	 */
 	protected function get_backend_address() {
-		$to_country = isset( $_POST['country'] ) ? strtoupper( wc_clean( $_POST['country'] ) ) : false;
-		$to_state   = isset( $_POST['state'] ) ? strtoupper( wc_clean( $_POST['state'] ) ) : false;
-		$to_zip     = isset( $_POST['postcode'] ) ? strtoupper( wc_clean( $_POST['postcode'] ) ) : false;
-		$to_city    = isset( $_POST['city'] ) ? strtoupper( wc_clean( $_POST['city'] ) ) : false;
-		$to_street  = isset( $_POST['street'] ) ? strtoupper( wc_clean( $_POST['street'] ) ) : false;
+		$to_country = isset( $_POST['country'] ) ? strtoupper( wc_clean( $_POST['country'] ) ) : false;  // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$to_state   = isset( $_POST['state'] ) ? strtoupper( wc_clean( $_POST['state'] ) ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$to_zip     = isset( $_POST['postcode'] ) ? strtoupper( wc_clean( $_POST['postcode'] ) ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$to_city    = isset( $_POST['city'] ) ? strtoupper( wc_clean( $_POST['city'] ) ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$to_street  = isset( $_POST['street'] ) ? strtoupper( wc_clean( $_POST['street'] ) ) : false; // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
```

Comment:
jfyi, AFAIK these are not flagged by the scanner (at least not by QIT - PHPCS).

`wc_clean` is an allowed sanitizing function, similar to `sanitize_text_field`, and checking `isset` on an unsanitized value won't trigger the warning

Reply:
Thanks, I am checking specifically against this rule `<rule ref="WordPress.Security.ValidatedSanitizedInput.InputNotSanitized"/>` and I get
```
FILE: /var/www/wordpress/wp-content/plugins/woocommerce-services/classes/class-wc-connect-taxjar-integration.php
---------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
---------------------------------------------------------------------------------------------------------------------------------------------------
 692 | ERROR | Detected usage of a non-sanitized input variable: $_POST['country']
     |       | (WordPress.Security.ValidatedSanitizedInput.InputNotSanitized)
 693 | ERROR | Detected usage of a non-sanitized input variable: $_POST['state']
     |       | (WordPress.Security.ValidatedSanitizedInput.InputNotSanitized)
 694 | ERROR | Detected usage of a non-sanitized input variable: $_POST['postcode']
     |       | (WordPress.Security.ValidatedSanitizedInput.InputNotSanitized)
 695 | ERROR | Detected usage of a non-sanitized input variable: $_POST['city'] (WordPress.Security.ValidatedSanitizedInput.InputNotSanitized)
 696 | ERROR | Detected usage of a non-sanitized input variable: $_POST['street']
     |       | (WordPress.Security.ValidatedSanitizedInput.InputNotSanitized)
```
Does the QIT run the phpcs rule differently? Should I use the QIT to run instead of the phpcs against `<rule ref="WordPress.Security.ValidatedSanitizedInput.InputNotSanitized"/>`? Thanks for checking!

Reply:
Removing `phpcs:ignore` for these lines.